modules/discriminators.py

```diff
@@ -181,10 +182,13 @@ async def _compare_and_regenerate_gpt_factory(
         generations = await comparison_generator(
             agent, middleman_settings_copy, compare_and_regenerate_prompt_v1
         )
-        for generation in generations.outputs:
+        for generation in generations.outputs or []:
```

Comment:
Should an error be raised here instead, if `generations.outputs is None`?

Reply:
I think it's plausible for there to be no outputs and for us to want to try again, so seems correct to not raise an error.